atlasdb-jepsen-tests/src/jepsen/atlasdb/lock.clj

```diff
@@ -0,0 +1,108 @@
+(ns jepsen.atlasdb.lock
+  (:require [jepsen.atlasdb.timelock :as timelock]
+            [jepsen.checker :as checker]
+            [jepsen.client :as client]
+            [jepsen.generator :as gen]
+            [jepsen.tests :as tests]
+            [jepsen.os.debian :as debian]
+            [jepsen.util :refer [timeout]]
+            [knossos.history :as history]
+            [clojure.java.io :refer [writer]])
+    (:import com.palantir.atlasdb.jepsen.JepsenHistoryChecker)
+    (:import com.palantir.atlasdb.http.LockClient))
+
+(def lock-names ["bob" "sarah" "alfred" "shelly"])
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Client creation and invocations (i.e. locking, unlocking refreshing)
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+(defn nullable-to-string    [obj] (if (nil? obj) "" (.toString obj)))
+(defn create-token-store    [] (atom {}))
+(defn replace-token         [store key obj] (swap! store assoc key obj))
+(defn assoc-ok-value
+  "Associate the map with ':ok', meaning that the server response was 200.
+   Also capture the server response and the query parameters we used."
+  [op response query-params]
+  (assoc op :type :ok
+            :value (nullable-to-string response)
+            :query-params (nullable-to-string query-params)))
+
+(defn create-client
+  "Creates an object that implements the client/Client protocol.
+   The object defines how you create a lock client, and how to request locks from it. The first call to this
+   function will return an invalid object: you should call 'setup' on the returned object to get a valid one.
+  "
+  [lock-service, client-name, token-store]
+  (reify client/Client
+    (setup!
+      [this test node]
+      "Factory that returns an object implementing client/Client"
+        (create-client
+          (LockClient/create '("n1" "n2" "n3" "n4" "n5"))
+          (name node)
+          (create-token-store)))
+
+    (invoke!
+      [this test op]
+      "Run an operation on our client"
+      (timeout (* 30 1000)
+          (assoc op :type :fail :error :timeout)
+          (try
+            (let [lock-name (:value op)
+                  token-store-key (str client-name lock-name)]
```

Comment:
@JacekLach - I've just had a thought. I'm not actually sure it is necessary to include the client name in the key - isn't there already a unique map per client? Just wanted to check with you that I'm understanding the atom correctly.